benches/benches/set/blockchain.rs

```diff
@@ -108,17 +243,258 @@ pub fn run(c: &mut Criterion) {
+            .expect("failed to prepare contract");
+        input.post_call.extend(vec![
+            Opcode::gtf(0x16, 0x00, GTFArgs::ScriptData),
+            Opcode::MOVI(0x15, 2000),
+            Opcode::ALOC(0x15),
+            Opcode::ADDI(0x14, REG_HP, 1),
+        ]);
+        run_group_ref(&mut c.benchmark_group("croo"), "croo", input);
+    }
+
+    run_group_ref(
+        &mut c.benchmark_group("flag"),
+        "flag",
+        VmBench::new(Opcode::FLAG(0x10)),
+    );
+
+    run_group_ref(
+        &mut c.benchmark_group("gm"),
+        "gm",
+        VmBench::contract(rng, Opcode::GM(0x10, 1)).unwrap(),
+    );
+
+    let mut smo = c.benchmark_group("smo");
+
+    for i in linear.clone() {
+        let mut input = VmBench::contract(rng, Opcode::SMO(0x15, 0x16, 0x17, 0x18))
+            .expect("failed to prepare contract");
+        input.outputs.push(Output::message(Address::zeroed(), 1));
+        let index = input.outputs.len() - 1;
+        input.post_call.extend(vec![
+            Opcode::gtf(0x15, 0x00, GTFArgs::ScriptData),
+            // Offset 32 + 8+ 8 + 32
+            Opcode::ADDI(0x15, 0x15, 32 + 8 + 8 + 32),
+            Opcode::MOVI(0x16, i as Immediate18),
+            Opcode::MOVI(0x17, index as Immediate18),
+            Opcode::MOVI(0x18, 10),
+        ]);
+        input.data.extend(
+            Address::new([1u8; 32])
+                .iter()
+                .copied()
+                .chain(vec![2u8; i as usize]),
+        );
+        let predicate = Opcode::RET(REG_ONE).to_bytes().to_vec();
+        let owner = Input::predicate_owner(&predicate);
+        let coin_input = Input::coin_predicate(
+            Default::default(),
+            owner,
+            Word::MAX,
+            AssetId::zeroed(),
+            Default::default(),
+            Default::default(),
+            predicate,
+            vec![],
+        );
+        input.inputs.push(coin_input);
+        smo.throughput(Throughput::Bytes(i));
+        run_group_ref(&mut smo, format!("{i}"), input);
+    }
+
+    smo.finish();
+
+    let mut srwq = c.benchmark_group("srwq");
+
+    for i in linear.clone() {
+        let mut key = Bytes32::zeroed();
+
+        key.as_mut()[..8].copy_from_slice(&i.to_be_bytes());
+        let data = key.iter().copied().collect::<Vec<_>>();
+
+        let post_call = vec![
+            Opcode::MOVI(0x16, i as Immediate18),
+            Opcode::MOVI(0x17, 2000),
+            Opcode::MOVE(0x15, 0x16),
+            Opcode::MULI(0x15, 0x15, 32),
+            Opcode::ADDI(0x15, 0x15, 1),
+            Opcode::ALOC(0x15),
+            Opcode::ADDI(0x14, REG_HP, 1),
+        ];
+        let mut bench = VmBench::contract(rng, Opcode::SRWQ(0x14, 0x11, 0x27, 0x16))
+            .expect("failed to prepare contract")
+            .with_post_call(post_call)
+            .with_prepare_db(move |mut db| {
+                db.merkle_contract_state_insert(&contract, &key, &key)?;
```

Comment:
```suggestion
                let values = vec![key; i as usize];
                db.merkle_contract_state_insert_range(&contract, &key, &values)?;
```

Reply:
we weren't actually reading more than a single slot before, and the benchmark dependent param was just testing how fast it could translate unset slots into zeroed out vm memory.